sdk/python/kfp/components/_components.py

```diff
@@ -239,15 +240,10 @@ def expand_command_part(arg): #input values with original names
                 return None
             if isinstance(arg, (str, int, float, bool)):
                 return str(arg)
-            elif isinstance(arg, dict):
-                if len(arg) != 1:
-                    raise ValueError('Failed to parse argument dict: "{}"'.format(arg))
-                (func_name, func_argument) = list(arg.items())[0]
-                func_name=func_name.lower()
-
-                if func_name == 'value':
-                    assert isinstance(func_argument, str)
-                    port_name = func_argument
+            else:
```

Comment:
You don't need "else" here since the above if returns anyway. That way we can reduce one level nesting for the content in the else block.

Reply:
I can remove it. The reason I left the indent is just so that I do not introduce more code changes due to changed indentation.

Reply:
Done.

Reply:
It immediately caused a conflict =)